database/query/entities.sql

```diff
@@ -76,6 +76,13 @@ LIMIT 1;
 SELECT * FROM entity_instances
 WHERE entity_instances.entity_type = $1 AND entity_instances.project_id = ANY(sqlc.arg(projects)::uuid[]);
 
+-- GetEntitiesByProvider retrieves all entities of a given provider.
+-- this is how one would get all repositories, artifacts, etc. for a given provider.
+
+-- name: GetEntitiesByProvider :many
```

Comment:
Sure, we can! I regret not having an index for provider + project.

Reply:
oh! Yeah, this is going to get tricky since a provider can live in a parent project, and the entities originating from that provider can be in subprojects. Now I remember why this was done without the ProjectID as part of the query.

Reply:
should we pass a list of projects that are subprojects of the provider?
(I am guilty of removing the project ID from the getByID call as well thinking that "well, the ID is unique" and forgot about that potential issue as well)